api/base/exceptions.py

```diff
@@ -45,3 +45,13 @@ class Gone(APIException):
 class InvalidFilterError(ParseError):
     """Raised when client passes an invalid filter in the querystring."""
     default_detail = 'Querystring contains an invalid filter.'
+
+
+class UnconfirmedAccountError(APIException):
+    status_code = 400
+    default_detail = 'Please confirm your account before using the API.'
+
+
+class DeactivatedAccountError(APIException):
+    status_code = 410
```

Comment:
If it's the login request, it might not be 410. However, if you're doing CRUD on a deactivated user, then 410 seems appropriate.